src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/MTEIndustrialDehydrator.java

```diff
@@ -225,22 +226,36 @@ protected OverclockCalculator createOverclockCalculator(@NotNull GTRecipe recipe
 
     @Override
     public void onModeChangeByScrewdriver(ForgeDirection side, EntityPlayer aPlayer, float aX, float aY, float aZ) {
-        mDehydratorMode = !mDehydratorMode;
-        String aMode = mDehydratorMode ? "Dehydrator" : "Vacuum Furnace";
-        PlayerUtils.messagePlayer(aPlayer, "Mode: " + aMode);
-        mLastRecipe = null;
+        setMachineMode(nextMachineMode());
+        PlayerUtils.messagePlayer(
+            aPlayer,
+            String.format(StatCollector.translateToLocal("GT5U.MULTI_MACHINE_CHANGE"), getMachineModeName()));
     }
 
     @Override
-    public void saveNBTData(NBTTagCompound aNBT) {
-        super.saveNBTData(aNBT);
-        aNBT.setBoolean("mDehydratorMode", mDehydratorMode);
+    public boolean supportsMachineModeSwitch() {
+        return true;
+    }
+
+    @Override
+    public void setMachineModeIcons() {
+        machineModeIcons.clear();
+        machineModeIcons.add(GTUITextures.OVERLAY_BUTTON_MACHINEMODE_STEAM);
+        machineModeIcons.add(GTUITextures.OVERLAY_BUTTON_MACHINEMODE_LPF_FLUID);
+    }
+
+    @Override
+    public String getMachineModeName() {
+        return StatCollector.translateToLocal("GT5U.GTPP_MULTI_INDUSTRIAL_DEHYDRATOR.mode." + machineMode);
     }
 
     @Override
     public void loadNBTData(NBTTagCompound aNBT) {
+        // Migrates old NBT tag to the new one
+        if (aNBT.hasKey("mDehydratorMode")) {
+            machineMode = (aNBT.getInteger("mDehydratorMode") == 0) ? 0 : 1;
```

Comment:
This tag was saved previously as `boolean` so one should be read as `boolean`:
```suggestion
            machineMode = aNBT.getBoolean("mDehydratorMode") ? 1 : 0;
```
Also it would be very nice to have constants for each machine mode ([like so](https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/MTEIndustrialCuttingMachine.java#L61)) for better readability.